### PR TITLE
Validate staking contract keys when nodes register

### PR DESCRIPTION
**Pull request by joshuahannan**
Closes https://github.com/dapperlabs/flow-go/issues/5597

Adds code to validate that the staking key and networking key are valid public keys before allowing the node to be created and registered

TODO: Find out a way to get CI to pass. It currently can't because it needs to be able to use a pre-compiled version of relic

### CODE REVIEW

**Comment by joshuahannan** [comment]
@tarakby I don't remember what the conclusion was on getting relic working with CI here. Do you know what we can do to get it working?

**Review by tarakby — COMMENTED** [review]
Looks good to me! 

There are 3 more updates we can do:
 - remove the length check of the public keys (the 128 and 192 length checks).
 - add comments about the upcoming staking field `Proof of Possession (PoP) of the staking private key`. This can be added when we read the staking public key. 
  - add a TODO comment after to the `assert` check of the staking key saying "To verify the provided Proof of Possession of the staking private key"

File: contracts/FlowIDTableStaking.cdc [hunk 1]
```diff
@@ -167,6 +168,26 @@ pub contract FlowIDTableStaking {
                 !FlowIDTableStaking.getStakingKeyClaimed(key: stakingKey): "The stakingKey cannot have already been claimed"
             }
 
+            let stakeKey = PublicKey(
+                publicKey: stakingKey.decodeHex(),
+                signatureAlgorithm: SignatureAlgorithm.BLS_BLS12_381
+            )
+
+            assert(
+                stakeKey.isValid,
```
[the remark this answers is not in the document]
Comment (joshuahannan) [reply contracts/FlowIDTableStaking.cdc:177]: yes, we are asserting that the variable is `true`

**Review by jordanschalm — COMMENTED** [review]
👍  Looks good to me. Agree with Tarak's suggestion to add the PoP note

**Comment by joshuahannan** [comment]
@tarakby is it really necessary to remove the length checks? I know that the key validation already does it, but it might be helpful to have the length checks anyway in case the user includes a `0x` or something like that by accident.

**Comment by tarakby** [comment]
> is it really necessary to remove the length checks? I know that the key validation already does it, but it might be helpful to have the length checks anyway in case the user includes a 0x or something like that by accident

@joshuahannan It is not necessary to remove it. This doesn't make the code more or less secure.
The idea was to make the contract agnostic of the crypto implementation of public keys (the crypto may update the serialization of keys for instance and those hardcoded lengths in the contract won't be valid). The `0x` would make the hex decode function fail anyway.

**Pull request merged**